src/Symfony/Component/Serializer/Normalizer/PropertyNormalizer.php

```diff
@@ -32,6 +36,38 @@
  */
 class PropertyNormalizer extends AbstractObjectNormalizer
 {
+    /**
+     * Flag to control whether public fields should be output when normalizing or omitted.
+     */
+    public const NORMALIZE_PUBLIC = 'normalize_public';
+
+    /**
+     * Flag to control whether protected fields should be output when normalizing or omitted.
+     */
+    public const NORMALIZE_PROTECTED = 'normalize_protected';
```

Comment:
I don't see any use case where we only want to omit protected properties.
IMHO, I'd rather set a "minimum visibility" enum context option with:
- `private`: to preserve all
- `protected`: to just omit private ones
- `public`: to just keep public properties

WDYT?

If you thing that use case is legit, why not using a bitmask instead? So that we only have one context option.

Reply:
I think keep possibility to compose visibility can be useful. I now use bitmask flags in order to allow that